pages/api/publish.js

```diff
@@ -2,47 +2,36 @@ import Ably from "ably/promises";
 import urlExists from "url-exists-nodejs";
 import ogs from "open-graph-scraper";
 
-const rest = new Ably.Rest(process.env.ABLY_SERVER_API_KEY);
```

Comment:
Simplified the control flow in this function by returning early when the http method is incorrect, or if the url is invalid. This makes the function much easier to read.